sql/expression/function/strcmp_test.go

```diff
@@ -36,10 +37,13 @@ func TestStrCmp(t *testing.T) {
 		{"equal strings", sql.Text, sql.Text, sql.NewRow("a", "a"), int(0), nil},
 		{"first string is smaller", sql.Text, sql.Text, sql.NewRow("a", "b"), int(-1), nil},
 		{"second string is smaller", sql.Text, sql.Text, sql.NewRow("b", "a"), int(1), nil},
-		{"arguments have different types", sql.Int8, sql.Text, sql.NewRow(1, "1"), int(0), nil},
 		{"first argument is null", sql.Text, sql.Text, sql.NewRow(nil, "a"), nil, nil},
 		{"second argument is null", sql.Text, sql.Text, sql.NewRow("a", nil), nil, nil},
 		{"both arguments are null", sql.Text, sql.Text, sql.NewRow(nil, nil), nil, nil},
+		{"different collations throws error", sql.MustCreateString(sqltypes.Text, 10, 255), sql.MustCreateString(sqltypes.Text, 10, 309), sql.NewRow("a", "a"), nil, sql.ErrCollationIllegalMix},
+		{"first argument is text, second argument is not text", sql.Text, sql.Date, sql.NewRow("a", 2022-11-04), int(1), nil},
```

Comment:
This looks a little misleading... `2022-11-04` looks like a date, but it really just resolves to the integer `2007`. Would be good to just change this to a single integer, without arithmetic operations, just so that its more obvious what is being tested.